lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3578,4 +3582,27 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Escapes metacharacters in a string intended to be used with a LIKE
+     * operator.
+     *
+     * @param string $inputString a literal, unquoted string
+     * @param string $escapeChar  should be reused by the caller in the LIKE
+     *                            expression.
+     */
+    final public function escapeStringForLike(string $inputString, string $escapeChar) : string
+    {
+        $replacePairs = [$escapeChar => $escapeChar . $escapeChar];
+        foreach (str_split($this->getLikeWildcardCharacters()) as $wildcardChar) {
```

Comment:
For RollerworksSearch I am using `addcslashes` is using str_split better for this?

Reply:
`$escapeChar` is not necessarily a backslash here, so no, `addcslashes` won't do

Reply:
`str_split` will not work with multibyte (Unicode) character, but would DB work with it?

Reply:
Not all dbs work with it, pg does, mysql does not

Reply:
I tried with 💩 the other day, no luck on mysql

Reply:
[addcslashes](http://php.net/addcslashes) also accepts a charlist 👍

Reply:
So, why do we need to smash all strings together and then decide how to split them back apart? What was wrong with the array?

Reply:
Personally I would avoid foreach entirely and just used preg_replace. Would work with single and multibyte as well.

Reply:
@Majkl578 you can build a regex from an array of inputs without having to split the string (I don't see a problem in using `foreach` or `array_map()` in this case).

It smells if you have to split/parse your own internal values.

Reply:
> addcslashes also accepts a charlist 👍

@sstok while I'm really glad to read that wonderful piece of news, I think it still does not let you pick what character you want for escaping, does it?

> So, why do we need to smash all strings together and then decide how to split them back apart? What was wrong with the array?

I'm not sure. What's good is that we now are sure we will only get one character.

> Personally I would avoid foreach entirely and just used preg_replace. Would work with single and multibyte as well.

Should I add support for using 💩 even though all RDBMs do not support that?

I'm a bit undecided on the array vs string debate. Regarding `preg_replace`, I'm not sure it makes a lot of sense here, because `str_replace` would do the same job quite well (right?)

Reply:
> Jokes aside, but why would anyone want to use a multibyte character for escaping.

can't see a reasonable usage

> Pracically, I don't see any difference between them. Unless performance counts which I doubt. Anyways, you'll have to build either the search and replace pieces, or the regex from the raw input of characters which I still prefer to be defined individually.

Yeah performance won't be an issue here IMO. Typical usage of this will be once or twice per http request. Reverting to arrays then

Reply:
I still think whole froreach + array of chars + strtr (+ single char validation) should be replaced just by something like:
```php
preg_replace('~([' . preg_quote($this->getLikeWildcardCharacters(), '~') . '])~u', addcslashes($escapeChar, '\\') .'$1', $inputString)
```

Reply:
@Majkl578 you can `implode()` the array of characters into a string like in your example.